### DISCUSSION TITLE
Integration of Remote control cable and protective tube in TEKSI modules

### DISCUSSION THREAD

**Discussion by sjib** [teksi/Home#100] [Ideas]
**Initial situation**
The [SIA405 Norm Geodaten zu Ver- und Entsorgungsleitungen](https://www.sia.ch/de/dienstleistungen/normen/themen/geodaten/) defines besides the media models for wastewater, water, gas, distant heating, electricty and communication also two extra models for 

- Remote control cable (or Telecontrol cable) (Fernwirkkabel / Cable de control à distance) (2 classes cable and cable_point) and 
- Protective tube (Schutzrohr / Tube de protection) (1 class protective_tube)

Fernwirkkabel (neuer Begriff für Signal- und Steuerungskabel) dienen der Datenübertragung von Betriebsinformationen (z.B. Messwerte) und der Fernsteuerung von Armaturen und Anlage für ein bestimmtes Medium. Fernwirkkabel werden in der Regel bei Rohrmedien (Abwasser, Fernwärme, Gas, Wasser usw.) modelliert.

Translation with DeepL
Telecontrol cables (new term for signal and control cables) are used for data transmission of operating information (e.g. measured values) and remote control of valves and systems for a specific medium. Telecontrol cables are usually modeled for pipe media (waste water, district heating, gas, water, etc.).

**Implementation in TEKSI**
There a two basic options to integrate these two extra modules:
a) Integrate **in schema of main module** (TEKSI water, TEKSI wastewater etc.)

- add the three extra classes in the main module schema with e.g. tww_cable, tww_cable_point and tww_protective_tube

Advantages: 

- Extra two modules come along with module installation

Disadvantages:

- INTERLIS Import / Export has to be defined several times with different schema name
- Maintenance of module is trickier, as same changes have to be adapted for each module

b) Have a central **extra schema for those two modules** that can be integrated in the main module database

- create an extra schema tcp (TEKSI (Remote) Cable and Protection tube) with the three extra classes in the main module schema with tcp _cable, tcp _cable_point and tcp _protective_tube
- Add an extra TEKSI attribute to define the media a class belongs to

Advantages: 

- The same schema can be used by different modules
- INTERLIS Import / Export has to be defined only once (always same schema name)

Disadvantages:

- Extra modules have to be installed in a extra step

At all - do you have any other aspects and arguments for either option? Please comment.

**Comment by ponceta** [comment]
- [ ] txx_od.sia405pt_protective_tube 
- [ ] txx_od.sia405cc_cable
- [ ] txx_od.sia405cc_cable_point
